### PR TITLE
fix: #265 and #261 (`/js/ldavis.v3.0.0.js`)

### LINKED ISSUE #265: major LDAviz UI topic index bug (e.g. term, topic, etc.)
I believe this is a high impact bug as it is not easy to notice, and can lead to incorrect interpretation, especially item 5.

-------

## Steps to reproduce

As of `pyldavis==3.4.1`, the bug should manifest itself as a minor UI bug (`item 2,3`), and data inconsistency (`item 5`).
It can be reproduced as follow:
1. prepare a visualisation with `start_index=0`
2. click topic 0
3. hover other topic and observe that topic will never be set to color1 (~blue) again, stuck at color2 (~red)
4. clicking other topic and observe same behaviour
5. hover a term and observer radius of topic does not agree with data, in fact, they are off by 1.

-------

## Blame

This commit: https://github.com/bmabey/pyLDAvis/pull/198/commits/010cde01e60920549d7305480110e8e1d9cdb139#diff-12750ebcae02c096cb6939beb54d0ab2353800555b8c01b00fae4fe93f376cfd

reverted some changes that are necessary for non-zero indexing to be functional:
https://github.com/bmabey/pyLDAvis/commit/f1cdb6521e45228c2f059e4a232118b767c34fe3#diff-ce28a7d983569ace041957292840319d65e2c6352989ae5f5556ae555e89404d

-------

## Solutions

The quickest and simplest fix is by reverting changes to all `start_index` related logic.

However, in the long run, we should probably decouple indexing logic with Topic labels, which will then allow renaming topics, which is sometimes requested by users.

### PR DESCRIPTION
**Pull request by ed9w2in6**
fixes #265 and #261

There does not seems to be any testing mechanism for the javascript code so no test are added.
Tested locally myself, matches my own expectation as described in #265.

@bmabey Please check.

### CODE REVIEW

**Comment by ed9w2in6** [comment]
@msusol

**Comment by ed9w2in6** [comment]
Also fixes #261, which just changes the formula written in the footnote of the visualisation.
For details please refer to #261.

**Comment by ed9w2in6** [comment]
@msusol @bmabey Hi, I would love to see this being merged since apparently in modern browsers Javascript files cannot be served without a server.  This means I cannot use this fix myself.

Let me know if you would need any help.

Of course, in the long run we should decouple the numbering from labels such that feature request like #92 can be implemented.  Decoupling should also remove the need of `sort_topics`, and `start_index`.

File: pyLDAvis/js/ldavis.v3.0.0.js [hunk 1]
```diff
@@ -178,7 +182,7 @@ var LDAvis = function(to_select, data_or_file_name, color1, color2) {
                 if (termElem !== undefined) term_off(termElem);
                 vis_state.term = "";
                 var value_old = document.getElementById(topicID).value;
-                var value_new = Math.min(K, +value_old + 1).toFixed(0);
+                var value_new = Math.min(K - 1 + startIndex, +value_old + 1).toFixed(0);
```
Comment (msusol) [inline pyLDAvis/js/ldavis.v3.0.0.js:185]: `K - 1 + startIndex` -> `K + startIndex - 1`

Just to be consistent everywhere

File: pyLDAvis/js/ldavis.v3.0.0.js [hunk 2]
```diff
@@ -211,8 +215,9 @@ var LDAvis = function(to_select, data_or_file_name, color1, color2) {
                 vis_state.term = "";
                 topic_off(document.getElementById(topicID + vis_state.topic));
                 var value_new = document.getElementById(topicID).value;
-                if (!isNaN(value_new) && value_new > 0) {
-                    value_new = Math.min(K, Math.max(1, value_new));
+                // older versions check for value_new !== "" too, why?
+                if (!isNaN(value_new) && value_new > startIndex - 1) {
+                    value_new = Math.min(K + startIndex - 1, Math.max(startIndex, value_new));
```
Comment (msusol) [inline pyLDAvis/js/ldavis.v3.0.0.js:220]: see above, match this

File: pyLDAvis/js/ldavis.v3.0.0.js [hunk 3]
```diff
@@ -512,7 +517,7 @@ var LDAvis = function(to_select, data_or_file_name, color1, color2) {
             .attr("x", 0)
             .attr("y", mdsheight + 10 + (8/2)*barguide.height + 5)
             .style("dominant-baseline", "middle")
-            .text("2. relevance(term w | topic t) = \u03BB * p(w | t) + (1 - \u03BB) * p(w | t)/p(w); see Sievert & Shirley (2014)");
+            .text("2. relevance(term w | topic t; \u03BB) = \u03BB * log(p(w | t)) + (1 - \u03BB) * log(p(w | t)/p(w)); see Sievert & Shirley (2014)");
```
Comment (msusol) [inline pyLDAvis/js/ldavis.v3.0.0.js:520]: https://github.com/bmabey/pyLDAvis/issues/261#issuecomment-2062986867

You don't mention this formula change?

Comment (msusol) [inline pyLDAvis/js/ldavis.v3.0.0.js:520]: Why is this change here?

`relevance(term w | topic t; \u03BB)`

**Review by msusol — CHANGES_REQUESTED** [review]
Please address these comments to pull in the changes.

File: pyLDAvis/js/ldavis.v3.0.0.js (hunk 3 above)
Comment (ed9w2in6) [reply pyLDAvis/js/ldavis.v3.0.0.js:520]: The notation is to indicate that Lambda is a parameter of the distribution, such that the formula aligns better to the one in the linked paper:
image

Comment (ed9w2in6) [reply pyLDAvis/js/ldavis.v3.0.0.js:520]: I changed it at line 511.  I am not sure what you mean.

File: pyLDAvis/js/ldavis.v3.0.0.js (hunk 1 above)
Comment (ed9w2in6) [reply pyLDAvis/js/ldavis.v3.0.0.js]: added commit 0a965059a9fca360e4e4ea212d88fd8657327bed

File: pyLDAvis/js/ldavis.v3.0.0.js (hunk 2 above)
Comment (ed9w2in6) [reply pyLDAvis/js/ldavis.v3.0.0.js:220]: added commit 0a965059a9fca360e4e4ea212d88fd8657327bed
